### PR TITLE
ZEN-3440 - QuickOutline and external preferences: filter files by content type

### PR DESCRIPTION
**Pull request by tfesenko**
QuickOutline and code-assist for references can show elements from v2 models for v3 and vice versa, see https://modelsolv.atlassian.net/browse/ZEN-3509?focusedCommentId=40220

### CODE REVIEW

**Comment by tfesenko** [comment]
@andylowry , can you please review and QA?

File: com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/assist/JsonContentAssistProcessor.java [hunk 1]
```diff
@@ -85,9 +85,9 @@
 
     private String[] textMessages;
     
-	public JsonContentAssistProcessor(ContentAssistant ca) {
+	public JsonContentAssistProcessor(ContentAssistant ca, String fileContentType) {
 		this(ca, new JsonProposalProvider(),
```
Comment (tfesenko) [inline com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/assist/JsonContentAssistProcessor.java]: Test comment

File: com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerContentDescriber.java [hunk 2]
```diff
@@ -13,6 +13,8 @@
 import com.reprezen.swagedit.core.editor.TextContentDescriber;
 
 public class SwaggerContentDescriber extends TextContentDescriber {
+	
+	public static final String CONTENT_TYPE_ID = "com.reprezen.swagedit.contenttype.swagger.yaml";
 
     @Override
     protected boolean isSupported(String content) {
```
Comment (andylowry) [inline com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerContentDescriber.java:20]: @tfesenko Maybe we should do a regex match to be more flexible regarding whitespace following the colons in the tests in `isSupported`?

**Review by andylowry — APPROVED** [review]

File: com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/editor/OpenApi3ContentDescriber.java [hunk 3]
```diff
@@ -13,6 +13,8 @@
 import com.reprezen.swagedit.core.editor.TextContentDescriber;
 
 public class OpenApi3ContentDescriber extends TextContentDescriber {
+	
+	public static final String CONTENT_TYPE_ID = "com.reprezen.swagedit.contenttype.openapi3.yaml";
 
     @Override
     protected boolean isSupported(String content) {
```
Comment (andylowry) [inline com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/editor/OpenApi3ContentDescriber.java:20]: @tfesenko Same as I later mention wrt 2.0 tests

**Comment by andylowry** [comment]
@tfesenko Code looks good, but it might make sense to be a little more lenient in content type checks, as noted above.
 
I'll QA in upcoming integration build.

**Comment by andylowry** [comment]
@tfesenko QA in integration build passed. No mixing up of content types that I could spot.

**Comment by tfesenko** [comment]
Replaced by #309

**Pull request closed without merging**